lib/auth/authentication_backend.rb

```diff
@@ -136,16 +138,31 @@ def get_group(group_name:)
       user_pool_id: user_pool_id
     )
   rescue Aws::CognitoIdentityProvider::Errors::ResourceNotFoundException => e
-    raise UserGroupNotFoundException.new(e.message)
+    raise UserGroupNotFoundException.new(e)
   rescue Aws::CognitoIdentityProvider::Errors::ServiceError => e
-    raise AuthenticationBackendException.new(e.message)
+    raise AuthenticationBackendException.new(e)
   end
 
   def authentication_backend_status
     client.describe_user_pool(user_pool_id: user_pool_id)
     OK
   rescue Aws::CognitoIdentityProvider::Errors::ServiceError => e
-    raise AuthenticationBackendException.new(e.message)
+    raise AuthenticationBackendException.new(e)
+  end
+
+  def backend_change_password(proposed:, old:, access_token:)
```

Comment:
```suggestion
  def change_password(proposed:, old:, access_token:)
```
do we need to prepend the `backend_` word? .. staying consistent with other methods

Reply:
yes... Because the authentication backend method gets masked by the one in the controller.

Reply:
you mean you have a name clash? I would then rename the method in the controller to be `change` (or anything else) or call this method explicitly `AuthenticationBackend.change_password`